packages/subgraph/src/v1-analytics/mapping.ts

```diff
@@ -291,8 +332,52 @@ export function handleTransactionCancelled(event: TransactionCancelled): void {
       assetBalance.assetId = transaction.receivingAssetId;
       assetBalance.router = event.params.router.toHex();
       assetBalance.amount = new BigInt(0);
+      assetBalance.supplied = new BigInt(0);
+      assetBalance.locked = new BigInt(0);
     }
     assetBalance.amount = assetBalance.amount.plus(transaction.amount);
+    assetBalance.locked = assetBalance.locked.sub(transaction.amount);
     assetBalance.save();
   }
+
+  // update metrics
+  let timestamp = event.block.timestamp.toI32();
+
+  let hour = timestamp / 3600; // rounded
+  let hourStartTimestamp = hour * 3600;
+
+  let hourIDPerAsset = hour.toString() + "-" + transaction.receivingAssetId.toHex();
+
+  let hourlyMetric = HourlyMetric.load(hourIDPerAsset.toString());
+  if (hourlyMetric === null) {
+    hourlyMetric = new HourlyMetric(hourIDPerAsset.toString());
+    hourlyMetric.hourStartTimestamp = BigInt.fromI32(hourStartTimestamp);
+    hourlyMetric.assetId = transaction.receivingAssetId.toHex();
+    hourlyMetric.volume = BigInt.fromI32(0);
+    hourlyMetric.liquidity = BigInt.fromI32(0);
+    hourlyMetric.txCount = BigInt.fromI32(0);
+    hourlyMetric.sendingTxCount = BigInt.fromI32(0);
+    hourlyMetric.receivingTxCount = BigInt.fromI32(0);
+    hourlyMetric.cancelTxCount = BigInt.fromI32(0);
+  }
+
+  let day = timestamp / 86400; // rounded
+  let dayStartTimestamp = day * 86400;
```

Comment:
this math isn't right - we should be flooring down to the beginning of the day... right?
so
```ts
dayStartTimestamp = day - (timestamp % 86400);
```

Reply:
i think the timestamp is really just to show when we were able to start collecting this data. we can't really floor it down to the beginning of the day because this should be used to show how much of the day these metrics are missing (they are only updated as the subgraph gets new entities)

Reply:
I'm still missing something here. bc according to this `dayStartTimestamp === timestamp`

You are just dividing by 86400 then multiplying by 86400 again